### ISSUE TITLE
Update xUnit2013 to stop warning about .Count check with StringValues

### ISSUE THREAD

**Issue by aremes** [xunit/xunit#2859]
Hope this is appropriate here. Seems to be an issue with analyzers only, but there's no separate repo for them, right?

Affected version: 2.6.4

Actual behavior:
```csharp
Microsoft.Extensions.Primitives.StringValues v = Microsoft.Extensions.Primitives.StringValues.Empty;
Assert.True(v.Count == 0); // Message: xUnit2024 -> Tells me to use Assert.Equal. Makes sense
Assert.Equal(0, v.Count); // Warning: xUnit2013 -> Tells me to use Assert.Empty. Makes sense
Assert.Empty(v); // ArgumentNullException: Value cannot be null (Parameter 'value'). <Nullable>enabled</Nullable> gives me a "Possible null reference" though.
Assert.Null(v); // Warning: xUnit2002 -> Tells me i shouldnt check for null on value types. Makes sense.
Assert.Equal(Microsoft.Extensions.Primitives.StringValues.Empty, v); // Works. Not really intuitive though.
```
Expected behavior: Assert.Empty should correctly handle StringValues.Empty and not throw an Exception.

* Is this a question? **No**
* Is this a bug with VSTest**No**
* Did you search the docs @ https://xunit.net/ ? **Yes**
* Did you search the issues list to see if someone already reported it? **Yes** _Might_ be related to #1970 
* Did you create a simple repro for the problem? 
**Yes:** [XUnit.StringValues.zip](https://github.com/xunit/xunit/files/13840432/XUnit.StringValues.zip)

**Comment by bradwilson** [comment]
Yeah, that's a weird one. I'm not sure how `readonly struct` differs from `struct` here, but the fact that a struct can be `null` doesn't make sense to me. So time to investigate what I don't know. 😂

**Comment by bradwilson** [comment]
Oh! I get it. It's not calling the `Assert.Empty` you think it's calling:

![image](https://github.com/xunit/xunit/assets/16944/355f9469-64d7-48ed-bcdc-415ab62b7d0e)

This is because there is an implicit cast from `StringValues` to `string`, and it apparently has higher compiler precedence than treating it as `IEnumerable`, which is what you (reasonably) expected. (`Assert.Null` doesn't have this cast, because there are no overloads: everything is treated as an `object`.) And the thing you're getting bitten by here is obviously, `StringValues.Empty` when cast to a `string` is a `null` string (not an empty one).

There's nothing we can do about this, because the only solution I know of would be to add an overload for `StringValues`, but we cannot take that dependency so we can't actually do that. That could be something you choose to do in your own project, if you were willing to extend our assertion library.

**Comment by bradwilson** [comment]
Although we can't change the assertion-side problem, we could update xUnit2013 to stop complaining when you check against `.Count` if the collection is `StringValues`.

**Comment by bradwilson** [comment]
We could also consider adding an analyzer to warn you against using `Assert.Empty` with a `StringValues` object.

**Comment by bradwilson** [comment]
Available in `1.10.0-pre.3` https://xunit.net/docs/using-ci-builds
